From f011501e719e0042fc6ca29ac7b33e8a14562246 Mon Sep 17 00:00:00 2001
From: "mark.yang" <mark.yang@lge.com>
Date: Tue, 1 Apr 2025 19:45:01 +0900
Subject: [PATCH] fix cstdint error with gcc-15.0.1

* see more details:
  http://errors.yoctoproject.org/Errors/Details/850148/
    TOPDIR/tmp/work/core2-64-oe-linux/sqlite-orm/1.5/git/examples/synchronous.cpp:7:5: error: 'uint16_t' does not name a type
        7 |     uint16_t src_port;
        |     ^~~~~~~~
    TOPDIR/tmp/work/core2-64-oe-linux/sqlite-orm/1.5/git/examples/synchronous.cpp:3:1: note: 'uint16_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
        2 | #include <sqlite_orm/sqlite_orm.h>
    +++ |+#include <cstdint>
        3 | #include <string>

While this was backported from 6a96d1cec4(Explicitly included <cstdint> in examples and unit tests),
since it was a commit made in v1.9, only synchronous.cpp was modified as other files do not exist in the current version.

Signed-off-by: mark.yang <mark.yang@lge.com>

Upstream-Status: Backport [6a96d1c Explicitly included <cstdint> in examples and unit tests]
---
 examples/synchronous.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/examples/synchronous.cpp b/examples/synchronous.cpp
index d10da13d..b103b48c 100644
--- a/examples/synchronous.cpp
+++ b/examples/synchronous.cpp
@@ -1,15 +1,16 @@
 
 #include <sqlite_orm/sqlite_orm.h>
+#include <cstdint>
 #include <string>
 
 struct Query {
     std::string src_ip;
-    uint16_t src_port;
-    uint16_t txn_id;
-    uint32_t tv_sec;
-    uint32_t tv_usec;
+    std::uint16_t src_port;
+    std::uint16_t txn_id;
+    std::uint32_t tv_sec;
+    std::uint32_t tv_usec;
     std::string name;
-    uint16_t type;
+    std::uint16_t type;
 };
 
 int main(int, char **) {

